resolver/resolver.cc

```diff
@@ -2962,7 +2962,7 @@ class ResolveTypeMembersAndFieldsWalk {
                 if (!klass.data(ctx)->typeMembers().empty()) {
                     dependencies_.emplace_back(klass);
                 }
-            } else if (symbol.isTypeAlias(ctx)) {
+            } else if (symbol.isTypeAlias(ctx) || symbol.isTypeMember()) {
```

Comment:
The existing fixed point in ResultTypeMembersAndFieldsWalk (which we originally needed because of `<AttachedClass>` iirc) magically solves most of what I had thought would be the hard part of this.